src/screens/OtherProfile/hooks/useOtherProfileScreenHooks.tsx

```diff
@@ -17,8 +32,11 @@ const useOtherProfileScreenHooks = (targetUserProfileId: string, username: strin
   const [isBlocking, setIsBlocking] = React.useState<boolean | undefined>(undefined);
   const [otherProfileData, setOtherProfileData] = React.useState<any>({});
   const [, setOtherProfileUserId] = React.useState<string>('');
-
-  const {updateFollowDiscoveryContext} = useDiscovery();
+  const [other_id, setOtherId] = React.useState('');
+  const [isCurrentFollowed, setIsCurrentFollowed] = React.useState(
+    getIsMeFollowingTargetStatus(params.data.other_id, params?.data?.username) ||
+      params?.data?.following
+  );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The initialization of `isCurrentFollowed` state variable seems to be dependent on the `params?.data?.following` and the result of `getIsMeFollowingTargetStatus(params.data.other_id, params?.data?.username)`. This could potentially lead to an undefined state if `params?.data?.following` or `params.data.other_id` is undefined. It would be better to add checks for these values before using them.

```diff
- const [isCurrentFollowed, setIsCurrentFollowed] = React.useState(
-   getIsMeFollowingTargetStatus(params.data.other_id, params?.data?.username) ||
-     params?.data?.following
- );
+ let initialFollowStatus = false;
+ if (params?.data?.other_id && params?.data?.username) {
+   initialFollowStatus = getIsMeFollowingTargetStatus(params.data.other_id, params?.data?.username);
+ }
+ if (params?.data?.following !== undefined) {
+   initialFollowStatus = params?.data?.following;
+ }
+ const [isCurrentFollowed, setIsCurrentFollowed] = React.useState(initialFollowStatus);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->